### PR TITLE
Node install script

### PR DESCRIPTION
**Pull request by dacook**
### What? Why?

- related to https://github.com/openfoodfoundation/ofn-install/issues/838

This script executes `nodenv` to ensure that the correct version of Node is installed and ready to use. It should work wherever `nodenv` is installed according to official documentation.

This could be used on a dev or server environment. But for now I didn't end up using it in 
 ofn-install this time (https://github.com/openfoodfoundation/ofn-install/pull/866).

### What should a dev test?

**1. install current version**
```
script/nodenv-install.sh
```

**2. install new version**
```
# edit .node-version with a new version number first, eg 17.9.1
script/nodenv-install.sh
```

**3. install unknown version (test node-build)**
If there's an unknown version, it will update node-build in case it's a new version it didn't know about.
```
# edit .node-version with a non-existent version, like "blah"
script/nodenv-install.sh
```

### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### Documentation updates
The new script could be mentioned in the getting started guide.

### CODE REVIEW

File: script/nodenv-install.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#
+# If our node-build version is outdated and it can't build the version we want
+# then we try upgrading node-build and installing again.
+
+if nodenv install --skip-existing; then
+  echo "Correct Node version is installed."
+else
+  echo "Upgrading node-build:"
+  # If homebrew (macOS) installed, try that first. Otherwise look in plugins directory.
+  brew upgrade node-build || git -C "$(nodenv root)"/plugins/node-build pull
```
Comment (mkllnk) [inline script/nodenv-install.sh:13]: I had a quick test. It's all good but I didn't like to see:

```
Upgrading node-build:
./script/nodenv-install.sh: line 13: brew: command not found
```

So I have a suggestion:

```suggestion

  if command -v brew &> /dev/null; then
    # Installation via Homebrew is recommended on macOS.
    brew upgrade node-build
  else
    git -C "$(nodenv root)"/plugins/node-build pull
  fi
```

**Review by mkllnk — APPROVED** [review]

File: script/nodenv-install.sh (hunk 1 above)
Comment (dacook) [reply script/nodenv-install.sh]: Oh, thank you, I should have seen that!

I've not seen `command -v` before. 
I just did a quick comparison with `which` out of interest, and found it performed the same for me (return `0` error code if available and `1` error code if not)

Comment (mkllnk) [reply script/nodenv-install.sh]: Here's some information about `command` being faster and safer: https://stackoverflow.com/a/677212/3377535

File: script/nodenv-install.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#
+# If our node-build version is outdated and it can't build the version we want
+# then we try upgrading node-build and installing again.
+
+if nodenv install --skip-existing; then
+  echo "Correct Node version is installed."
+else
+  echo "Upgrading node-build:"
+
+  if command -v brew &> /dev/null; then
```
Comment (jibees) [inline script/nodenv-install.sh:13]: I didn't know about that `command`, which is pretty convenient here. 
I used to use `if [[ "`uname`" == "Darwin" ]]; then` to test if current is Mac:

https://github.com/openfoodfoundation/openfoodnetwork/blob/ace74dc42d8855c908c7ce7bfb8056609c466718/script/rspec-slow-repeat#L33-L37

**Review by jibees — APPROVED** [review]
👍

**Comment by dacook** [comment]
I just tested this again (because .node-version got updated yay!) and it worked fine for me. 
@mkllnk did you already test this earlier on your Debian install?

**Comment by mkllnk** [comment]
> did you already test this earlier on your Debian install?

Yes.

**Comment by dacook** [comment]
👍 Thanks, it's reviewed, approved and tested. Merging.

**Pull request merged**